### PR TITLE
add initial EL section to FOCIL EIP

### PR DESCRIPTION
**Pull request by jacobkaufmann**
add initial execution layer section for FOCIL EIP.

### CODE REVIEW

File: EIPS/eip-FOCIL.md [hunk 1]
```diff
@@ -31,7 +31,25 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
 
 ### Execution Layer
 
-TBD
+On the execution layer, the block validity conditions are extended such that, after all of the transactions in the block have been executed, we attempt to execute each valid transaction in the inclusion list that was not present in the block.
```
Comment (soispoke) [inline EIPS/eip-FOCIL.md:34]: should we specify the validity of txns is a check on the nonce and balance for each sender?

**Review by soispoke — COMMENTED** [review]
lgtm, just one comment for clarification

File: EIPS/eip-FOCIL.md [hunk 2]
```diff
@@ -31,7 +31,25 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
...
+
+For each transaction `T` in the inclusion list, perform the following:
+
+1. Check whether `T` is present in `B`. If `T` is present, then continue to the next transaction.
+1. Validate `T`. If `T` is invalid, then continue to the next transaction.
+1. Execute `T` on state `S`. Assert that the execution of `T` fails.
+
+Note that we do not need to reset the state to `S`, since the only way for a transaction to alter the state is for it to execute sucessfully, in which case the block is invalid, and so the block will not be applied to the state.
+
+We make the following changes to the engine API:
```
Comment (jihoonsong) [inline EIPS/eip-FOCIL.md:48]: Do we also need a new engine API that allows the proposer's CL to notify the EL a set of local ILs to update its payload?

Comment (jacobkaufmann) [reply EIPS/eip-FOCIL.md:48]: are you suggesting an engine API that would incrementally notify the EL of one or more local inclusion lists as those messages arrive over the p2p network?

currently, with `engine_forkchoiceUpdated`, the latest aggregate inclusion list would be included in the payload attributes.

File: EIPS/eip-FOCIL.md (hunk 1 above)
Comment (jacobkaufmann) [reply EIPS/eip-FOCIL.md:34]: do we need to specify that a transaction might be invalid before any IL transactions execute but may become valid via execution of one or more transactions in the IL?

File: EIPS/eip-FOCIL.md (hunk 2 above)
Comment (jihoonsong) [reply EIPS/eip-FOCIL.md:48]: Yes I thought it'd be necessary if the proposer initiated block building process before it gathers all inclusion lists it needs. The proposer won't get the full list until 9s at least. When is the block building process begin? Does it usually happen much later than 9s?

Comment (jihoonsong) [reply EIPS/eip-FOCIL.md:48]: Let me elaborate what I'm trying to ask. If we only use `engine_forkchoiceUpdated` for notifying the EL about the ILs, the CL has no choice but to call it to initiate payload building after 11s. But if we have another new engine API something like `engine_updatePayloadWithInclusionLists` that allows the CL to inform the EL about the ILs, the proposer can start payload building anytime it wants and just let it comply with IL constraints at 11s by calling the new engine API. Please let me know if I missed anything!

Comment (jacobkaufmann) [reply EIPS/eip-FOCIL.md:48]: I'm not sure if the timing is specified explicitly in the `consensus-specs`.

there is the following note in the `bellatrix` validator guide:

> It is recommended for a validator to call `prepare_execution_payload` as soon as input parameters become known, and make subsequent calls to this function when any of these parameters gets updated.

following that, it seems okay to only have `engine_forkchoiceUpdated` called once in expectation, with subsequent calls possible by changing the inclusion list parameter in the payload attributes.

**Comment by soispoke** [comment]
lgtm, thanks!

**Pull request merged**